### PR TITLE
Mesh Render Extent Settings

### PR DESCRIPTION
**Pull request by JanCaha**
## Description

This is iteration of #59305 which was changed a lot by introduction of changes in #59444. 

This PR brings to renderer for Mesh Contours option to set renderer Min Max based on extent (specific canvas, changing canvas) in the same way as renderers for Raster Layer.

Besides that minimumMaximumActiveScalarDataset( const QgsRectangle &extent, const QgsMeshDatasetIndex &datasetIndex, double &min, double &max ) is added to MeshLayer to obtain Min Max from specific Scalar Dataset Index for given extent.

396460968-2db9cad8-ae86-4bfc-9b4b-f493926bb8f5

### CODE REVIEW

File: src/core/mesh/qgsmeshrenderersettings.h [hunk 1]
```diff
@@ -115,6 +114,17 @@ class CORE_EXPORT QgsMeshRendererScalarSettings
       NeighbourAverage,
     };
 
+    /**
+     *  This enumerator describes the extent used to compute min/max values
+     */
+    enum class MinMaxValueType
```
Comment (nyalldawson) [inline src/core/mesh/qgsmeshrenderersettings.h:120]: Can you synchronize the naming here with those used in https://github.com/qgis/QGIS/pull/59644, and move this enum to Qgis accordingly?

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
LGTM! @wonder-sk did you want to review this one too?

**Comment by wonder-sk** [comment]
I think @uclaros will have a glance on this...

File: src/core/mesh/qgsmeshrenderersettings.cpp [hunk 2]
```diff
@@ -186,6 +196,69 @@ void QgsMeshRendererScalarSettings::setEdgeStrokeWidthUnit( Qgis::RenderUnit edg
...
+{
+  if ( extent == "WholeMesh" )
+  {
+    return Qgis::MeshRangeExtent::WholeMesh;
+  }
+  else if ( extent == "CurrentCanvas" )
+  {
+    return Qgis::MeshRangeExtent::FixedCanvas;
+  }
+  else if ( extent == "UpdatedCanvas" )
```
Comment (uclaros) [inline src/core/mesh/qgsmeshrenderersettings.cpp:233]:
```suggestion
  if ( extent == QLatin1String( "WholeMesh" ) )
  {
    return Qgis::MeshRangeExtent::WholeMesh;
  }
  else if ( extent == QLatin1String( "CurrentCanvas" ) )
  {
    return Qgis::MeshRangeExtent::FixedCanvas;
  }
  else if ( extent == QLatin1String( "UpdatedCanvas" ) )
```

File: src/gui/mesh/qgsmeshrendererscalarsettingswidget.cpp [hunk 3]
```diff
@@ -216,4 +265,66 @@ bool QgsMeshRendererScalarSettingsWidget::dataIsDefinedOnEdges() const
...
+    {
+      searchExtent = mMeshLayer->extent();
+      break;
+    }
+    case Qgis::MeshRangeExtent::FixedCanvas:
+    case Qgis::MeshRangeExtent::UpdatedCanvas:
+    {
+      QgsCoordinateTransform ct = QgsCoordinateTransform( mCanvas->mapSettings().destinationCrs(), mMeshLayer->crs(), QgsProject::instance() );
+      searchExtent = mCanvas->extent();
+      searchExtent = ct.transform( searchExtent );
```
Comment (uclaros) [inline src/gui/mesh/qgsmeshrendererscalarsettingswidget.cpp:291]: transform may throw a QgsCsException, we need to catch it!

**Review by uclaros — COMMENTED** [review]
Nice!
Just a couple of comments

**Pull request merged**